credentials/credentials_ext_test.go

```diff
@@ -175,6 +176,38 @@ func (s) TestAuthorityCallOptionWithInsecureCreds(t *testing.T) {
 	}
 }
 
+// Tests the scenario where the `grpc.CallAuthority` call option is used with
+// local transport credentials. The test verifies that the specified
+// authority is correctly propagated to the server.
+func (s) TestAuthorityCallOptionWithLocalCreds(t *testing.T) {
```

Comment:
Looking at this test *now*, it feels like there is some scope for making a table driven test for cases where the credentials is expected to accept the authority override, and the RPC is expected to succeed.

The table will have three entries: one for TLS, one for insecure and one for local. The table would also have fields for server and client credentials. 

And if you make a helper function that reads the server cert+key pair and the CA cert, and returns two `credentials.TransportCredentials`, one for the server and one for the client, you can populate the those two fields in the table (for the TLS case). For the other two cases, it would just be `insecure.NewCredentials` and `local.NewCredentials`.